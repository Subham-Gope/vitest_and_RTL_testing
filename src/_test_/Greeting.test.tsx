// import { render, screen, fireEvent } from "@testing-library/react";
// import { test, expect, vi } from "vitest";
// import { Greeting } from "../components/Greeting";

// test("calls onGreet with Alice", () => {
//   const mockGreet = vi.fn<(name: string) => void>();
//   render(<Greeting onGreet={mockGreet} />);

//   fireEvent.click(screen.getByText("Greet"));

//   expect(mockGreet).toHaveBeenCalledWith("Alice");
//   expect(mockGreet).toHaveBeenCalledTimes(1);
// });
