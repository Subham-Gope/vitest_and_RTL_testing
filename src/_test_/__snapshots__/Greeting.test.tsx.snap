// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`calls onGreet with Alice 1`] = `
<div>
  <div
    class="h-screen w-screen flex items-center justify-center gap-4"
  >
    <div
      class="w-[320px] px-8 py-12 border-3 border-gray-600 rounded-lg bg-amber-50 flex flex-col items-center jusitfy-center gap-4"
    >
      <h1
        class="text-2xl text-gray-800 text-center"
      >
        This is a test card
      </h1>
      <p
        class="text-lg text-center font-light"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus beatae doloribus, neque obcaecati labore error.
      </p>
      <button
        class="bg-amber-500 text-white px-12 py-2 rounded-[24px]"
      >
        Greet
      </button>
    </div>
  </div>
</div>
`;
